azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java

```diff
@@ -16,15 +16,121 @@
  */
 package org.jclouds.azurecompute.compute.functions;
 
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
 import org.jclouds.azurecompute.domain.OSImage;
 import org.jclouds.compute.domain.Image;
 
 import com.google.common.base.Function;
+import org.jclouds.compute.domain.ImageBuilder;
+import org.jclouds.compute.domain.OperatingSystem;
+import org.jclouds.compute.domain.OsFamily;
+import org.jclouds.domain.Location;
+import org.jclouds.domain.LocationBuilder;
+import org.jclouds.domain.LocationScope;
+import org.jclouds.location.suppliers.all.JustProvider;
+
+import static com.google.common.base.Preconditions.checkNotNull;
 
 public class OSImageToImage implements Function<OSImage, Image> {
+   private static final String UNRECOGNIZED = "UNRECOGNIZED";
+
+   private static final String UBUNTU = "Ubuntu";
+   private static final String WINDOWS = "Windows";
+   private static final String OPENLOGIC = "openLogic";
+   private static final String CENTOS = "CentOS";
+   private static final String COREOS = "CoreOS";
+   private static final String OPENSUSE = "openSUSE";
+   private static final String SUSE = "SUSE";
+   private static final String SQL_SERVER = "SQL Server";
+   private static final String ORACLE_lINUX = "Orcale Linux";
+
+   private final JustProvider provider;
+
+   @Inject
+   protected OSImageToImage(JustProvider provider) {
+      this.provider = provider;
+   }
 
    @Override
-   public Image apply(OSImage input) {
-      return null;
+   public Image apply(OSImage image) {
+
+      ImageBuilder builder = new ImageBuilder()
+            .id(image.name())
+            .name(image.label())
+            .description(image.description())
+            .status(Image.Status.AVAILABLE)
+            .uri(image.mediaLink())
+            .providerId(image.name())
+            .location(createLocation(image.location()));
+
+      OperatingSystem.Builder osBuilder = osFamily().apply(image);
+      return builder.operatingSystem(osBuilder.build()).build();
+   }
+
+   private Location createLocation(String input) {
+      if (input != null) {
+         return new LocationBuilder().id(input).scope(LocationScope.REGION).description(input).parent(
+               Iterables.getOnlyElement(provider.get())).metadata(ImmutableMap.<String, Object>of("name", input))
+               .build();
+      } else {
+         return null;
+      }
+   }
+
+   public static Function<OSImage, OperatingSystem.Builder> osFamily() {
+      return new Function<OSImage, OperatingSystem.Builder>() {
+         @Override
+         public OperatingSystem.Builder apply(final OSImage image) {
+
+            final String label = image.label();
+            checkNotNull(label, "label");
+            OsFamily family = OsFamily.UNRECOGNIZED;
+
+            if (label.contains(CENTOS))
+               family = OsFamily.CENTOS;
+            else if (label.toUpperCase().contains(OPENLOGIC))
```

Comment:
I'm a bit confused. Why the label is compared upper case in some cases and not everywhere?